detox/src/devices/Device.js

```diff
@@ -64,7 +56,7 @@ class Device {
     const _bundleId = bundleId || this._bundleId;
     if (this._isAppInBackground(params, _bundleId)) {
       if (hasPayload) {
-        await this.deviceDriver.deliverPayload({...params, delayPayload: true});
```

Comment:
`delayPayload` is needed for [iOS](https://github.com/wix/Detox/blob/d3a90729f50d5920371029d9e9debe85577f1047/detox/ios/Detox/DetoxManager.m#L269). Why was it removed?
As I said in our internal conversation, timing and these params are very important for the correct behavior of iOS, so please don't remove stuff.

Reply:
I originally added `delayPayload` for Android, and it was a bit of a hack. With `predeliver` its much cleaner, but nevertheless - iOS remains unaffected in both cases, so no worries!

Reply:
Hmm just spotted the native usage of `delayPayload` on iOS. Strikes me by surprise. I'm still looking into alternatives, then. In any case, let's not forget that this is the purpose of code reviews. Let's also try to figure out why this change hasn't broken any test - unit or e2e: this is a signal of that coverage is missing.

Reply:
It’s hard to test this in Detox E2E. The native is unit tested.